### ISSUE TITLE
Two quick questions: code blocks, iframes

### ISSUE THREAD

**Issue by amca01** [kaushalmodi/ox-hugo#410]
Hello,

I'm trying to resurrect a blog which has been sitting fallow for about a year.  In that time, Hugo ha moved on, and I'm finding some things not working the way they used to.  I also had to change my theme from Blackburn to Clarity, as Blackburn hasn't been updated for three years, and makes reference to deprecated Hugo functionality.  So:

**Code highlighting.** I always used pygments (setting various parameters in my config file), and would highlight a code block with

        #+begin_src python
        x = 2 + 2
        print(x)
        $+end_src
        
but this no longer seems to work; at least it's not "standard" Hugo from what I can tell.  The best result I've got so far was with
        
    #+begin_src
     // inserted code
    #+end_src

which certainly set off the code block, but with no syntax highlighting.  I tried all of

    ```python
    ````python
    ``````python

(with corresponding ending backticks); no help.  I've also tried using shortcodes, as recommended in the Hugo documentation.  Clearly there's a parameter I need to set somewhere... but what exactly I'm still haven't found.  (I've also tried setting `#+EXPORT_HUGO_CODE_FENCE: t` at the beginning of my org file.)

**iframes.**  Is there an approved method for including an iframe in a blog post?  I've tried including one in a source block

    #+begin_src html
    <iframe .... ></iframe>
    #+end_src

which didn't do the trick.  

I don't think these issues are to do with the new theme I've moved to; it should support standard Hugo (maybe with a few extras), and should certainly not prevent plain Hugo from being appropriately rendered.   But anyway I've been stuck on these for a while now and I'm getting very frustrated with myself - knowing all too well that the answers to both will be painfully simple.

Anyway, thanks very much!

cheers,
Alasdair

**Comment by kaushalmodi** [comment]
>I always used pygments

I haven't used pygments in many years, ever since Hugo switch to using Chroma as its default highlighter. I'd suggest that you switch to the default highlighter and then see what you get.

To highlight code fenced blocks, you need to set `pygmentsCodeFences = true` in your site's config.toml (yes, even when using Chroma).

Here's what I have:

```
pygmentsCodeFences = true       # This applies to Chroma too.
```

---

>  I've tried including one in a source block

>     #+begin_src html
>     <iframe .... ></iframe>
>     #+end_src

That should have worked .. if you have this in your site's config.toml:

```
[markup.goldmark.renderer]
  unsafe = true # This is *necessary* for ox-hugo to work well with Hugo 0.60.0+
```

See https://ox-hugo.scripter.co/doc/goldmark/#enable-unsafe-html .

**Comment by amca01** [comment]
Thank you!  I had indeed set `pygmentsCodeFences = true` in my configuration file, as well as `pygmentsUseClasses = false`.  At the very top of my org file containing all my posts I have `#+EXPORT_HUGO_CODE_FENCE: t`.  I had also set `unsafe=true` in the markup section of the config file.

I have only found one "light" highlight theme that works, which is monokailight.  I have no idea why I can't use any of the others.

However, with all of that set, the result of

    #+begin_src html
    <iframe ... ></iframe>
    #+end_src

is simply a highlighted rendering of the script `<iframe ... > </iframe>` as a line of code, rather than an insertion of the iframe itself.    

I have another quick question: what's the easiest way to create a link to another post in the same blog?

Thank you again - I'm just "getting back" into blogging after nearly a year away, and as usual wrestling with a myriad of issues, all mostly small, but still (at least for me!) tricky.  And as before I think ox-hugo is a blogger's dream.

cheers,
Alasdair

**Comment by benisjammin** [comment]
If you want to export literal html the org-mode syntax is
``` org
#+BEGIN_EXPORT html
<b>Insert unescaped html here to be exported literally.</b>
#+END_EXPORT
```
=>
```html
<b>Insert unescaped html here to be exported literally.</b>
```

`#+begin_src LANG` is for displaying code samples with syntax highlighting for LANG, so any html within the src-block is rendered as plain text. For example, if the following was found in an org file:
```org
#+begin_src html
<iframe ... > </iframe>
#+end_src

#+begin_export html
<iframe ... > </iframe>
#+end_export
```
ox-hugo will give the following markdown:
```md
```html
<iframe ... > </iframe>
```                                                                     # comment to preserve code block

<iframe ... > </iframe>

```

**Comment by amca01** [comment]
Yes: `#+begin_export ...  #+end_export`  for including iframes - but of course!  Face plant and forehead slap out of stupidity.  Works fine now.  Many thanks indeed!

**Comment by benisjammin** [comment]
If there both your questions are answered satisfactorily, you may want to close this issue, @amca01 .

**Comment by sprajagopal** [comment]
follow up: is there a way for ox-hugo to place the html file (in the iframe src tag) in the hugo static folder? As it does with images for example.

**Comment by kaushalmodi** [comment]
@amca01 I am closing this issue as your questions are resolved.

@benisjammin Many thanks for stepping in to answer those!

@sprajagopal I have moved your question to a separate issue.
